docs/gwcs/fits_analog.rst

```diff
@@ -0,0 +1,101 @@
+  CUNIT1  = 'deg'                / Units of coordinate increment and value
+  CUNIT2  = 'deg'                / Units of coordinate increment and value
+  CTYPE1  = 'RA---TAN'           / TAN (gnomonic) projection + SIP distortions
+  CTYPE2  = 'DEC--TAN'           / TAN (gnomonic) projection + SIP distortions
+  CRVAL1  =        5.63056810618 / [deg] Coordinate value at reference point
+  CRVAL2  =      -72.05457184279 / [deg] Coordinate value at reference point
+  LONPOLE =                180.0 / [deg] Native longitude of celestial pole
+  LATPOLE =      -72.05457184279 / [deg] Native latitude of celestial pole
+  RADESYS = 'ICRS'                / Equatorial coordinate system
+
+
+The following imports are generally useful:
+
+  >>> import numpy as np
+  >>> from astropy.modeling import models
+  >>> from astropy import coordinates as coord
+  >>> from astropy import units as u
+  >>> from gwcs import wcs
+  >>> from gwcs import coordinate_frames as cf
+
+The ``forward_transform`` is constructed as a combined model using `astropy.modeling`.
+The ``frames`` are subclasses of `~gwcs.coordinate_frames.CoordinateFrame`. Although strings are
+acceptable as ``coordinate_frames`` it is recommended this is used only in testing/debugging.
+
+Using the `~astropy.modeling` package create a combined model to transform
+detector coordinates to ICRS following the FITS WCS standard convention.
+
+First, create a transform which shifts the input  ``x`` and ``y`` coordinates by ``CRPIX``.  We subtract 1 from the CRPIX values because the first pixel is considered pixel ``1`` in FITS WCS:
+
+  >>> shift_by_crpix = models.Shift(-(2048 - 1)*u.pix) & models.Shift(-(1024 - 1)*u.pix)
+
+Create a transform which rotates the inputs using the ``PC matrix``.
+
+  >>> matrix = np.array([[1.290551569736E-05, 5.9525007864732E-06],
+  ...                    [5.0226382102765E-06 , -1.2644844123757E-05]])
+  >>> rotation = models.AffineTransformation2D(matrix * u.deg,
+  ...                                          translation=[0, 0] * u.deg)
+  >>> rotation.input_units_equivalencies = {"x": u.pixel_scale(1*u.deg/u.pix),
+  ...                                       "y": u.pixel_scale(1*u.deg/u.pix)}
+  >>> rotation.inverse = models.AffineTransformation2D(np.linalg.inv(matrix) * u.pix,
+  ...                                                  translation=[0, 0] * u.pix)
+  >>> rotation.inverse.input_units_equivalencies = {"x": u.pixel_scale(1*u.pix/u.deg),
+  ...                                               "y": u.pixel_scale(1*u.pix/u.deg)}
+
+Create a tangent projection and a rotation on the sky using ``CRVAL``.
+
+  >>> tan = models.Pix2Sky_TAN()
+  >>> celestial_rotation =  models.RotateNative2Celestial(5.63056810618*u.deg, -72.05457184279*u.deg, 180*u.deg)
+
+  >>> det2sky = shift_by_crpix | rotation | tan | celestial_rotation
+  >>> det2sky.name = "linear_transform"
+
+Create a ``detector`` coordinate frame and a ``celestial`` ICRS frame.
```

Comment:
Why do we need this `detector` frame? Can't we just bunch all transforms together from detector all the way to the sky? Also, what is a "detector frame"? is it just the value of the `name` argument in the `Frame2D` initializer?

Reply:
@nden comment should be added to the docs. My comment was intended to prompt a discussion/explanation about organization of the (gwcs) pipelines. For example, why do some pipelines have multiple steps (JWST WCS comes to mind)?

Reply:
I wonder if a paragraphs or two before this example outlining the general structure of a WCS object would be helpful. Any objection @nden?

Reply:
What if the current "Getting Started" is replaced by an earlier section:

```
Basic Structure of a GWCS Object
--------------------------------

The key concept to be aware of is that a GWCS Object consists of a pipeline
of steps; each step contains a transform (i.e., an Astropy model) that 
converts the input coordinates of the step to the output coordinates of
the step. Furthermore, each step has an optional coordinate frame associated
with the step. The coordinate frame represents the input coordinate frame, not
the output coordinates. Most typically, the first step coordinate frame is
the detector pixel coordinates (the default). Since no step has a coordinate 
frame for the output coordinates, it is necessary to append a step with no
transform to the end of the pipeline to represent the output coordinate frame.
For imaging, this frame typically references one of the Astropy standard
Sky Coordinate Frames of Reference. The GWCS frames also serve to hold the
units on the axes, the names of the axes and the physical type of the axis
(e.g., wavelength).

Since it is often useful to obtain coordinates in an intermediate frame of
reference, GWCS allows the pipeline to consist of more than one transform.
For example, for spectrographs, it is useful to have access to coordinates
in the slit plane, and in such a case, the first step would transform from
the detector to the slit plane, and the second step from the slit plane to
sky coordinates and a wavelength. Constructed this way, it is possible to
extract from the GWCS the needed transforms between identified frames of
reference.  

The GWCS object can be saved to the ASDF format using the
`asdf <https://asdf.readthedocs.io/en/latest/>`__ package and validated 
using `ASDF Standard <https://asdf-standard.readthedocs.io/en/latest/>`__

There are two ways to save the GWCS object to a files:

- `Save a WCS object as a pure ASDF file`_ 
```
@mcara does this address your comment?

Reply:
Yes! This is a great introduction.

Reply:
I like this too.

> Furthermore, each step has an optional coordinate frame associated with the step.

Is the frame optional? I thought it's required.

> a GWCS Object consists of a pipeline of steps

Suggest adding `linear` ->  `consists of a linear pipeline of steps`
so that it's clear it's not an arbitrary graph.